### PR TITLE
Use ssize_t and off_t definitions from sys/types.h

### LINKED ISSUE #37665: File system: wrong type for ssize_t in fs.h for CONFIG_ARCH_POSIX
**Describe the bug**
In fs.h ssize_t is defined as __SIZE_TYPE__, which is a unsigned type. ssize_t should be signed.

**To Reproduce**
Run/compile this code snipped, the printf statement will not be executed. 
#include "fs/fs.h"
..
ssize_t i = -10;
    if (i < 0) {
        printf("test ok");
    }

**Expected behavior**
printf will be executed.

**Impact**
Project unit tests on Linux are failing. 

**Logs and console output**

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK

**Additional context**
Proposal:
`
diff --git a/include/fs/fs.h b/include/fs/fs.h
index 76bf25bd10..6da866d58b 100644
--- a/include/fs/fs.h
+++ b/include/fs/fs.h
@@ -9,7 +9,7 @@
 
 #ifdef CONFIG_ARCH_POSIX
 #ifndef __ssize_t_defined
-typedef __SIZE_TYPE__ ssize_t;
+typedef __SWORD_TYPE ssize_t;
 #define __ssize_t_defined
 #endif
`

### PR DESCRIPTION
**Pull request by de-nordic**
The PR contains two commits:
 - removing fs.h local definitions of off_t ans ssize_t,, and using sys/types.h instead;
 - fixing littlefs sample printk formatting.

Fixes: #37665

### CODE REVIEW

File: samples/subsys/fs/littlefs/src/main.c [hunk 1]
```diff
@@ -149,9 +149,9 @@ void main(void)
 			printk("End of files\n");
 			break;
 		}
-		printk("  %c %u %s\n",
+		printk("  %c %lu %s\n",
 		       (ent.type == FS_DIR_ENTRY_FILE) ? 'F' : 'D',
-		       ent.size,
```
Comment (mniestroj) [inline samples/subsys/fs/littlefs/src/main.c:154]: "%zu" instead

File: samples/subsys/fs/littlefs/src/main.c [hunk 2]
```diff
@@ -73,8 +73,8 @@ void main(void)
 	!(FSTAB_ENTRY_DT_MOUNT_FLAGS(PARTITION_NODE) & FS_MOUNT_FLAG_AUTOMOUNT)
 	rc = fs_mount(mp);
 	if (rc < 0) {
-		printk("FAIL: mount id %u at %s: %d\n",
-		       (unsigned int)mp->storage_dev, mp->mnt_point,
+		printk("FAIL: mount id %llu at %s: %d\n",
+		       (unsigned long long)mp->storage_dev, mp->mnt_point,
```
Comment (stephanosio) [inline samples/subsys/fs/littlefs/src/main.c:77]:
```suggestion
		printk("FAIL: mount id %p at %s: %d\n",
		       mp->storage_dev, mp->mnt_point,
```

File: samples/subsys/fs/littlefs/src/main.c [hunk 3]
```diff
@@ -100,7 +100,7 @@ void main(void)
 	rc = fs_stat(fname, &dirent);
 	printk("%s stat: %d\n", fname, rc);
 	if (rc >= 0) {
-		printk("\tfn '%s' siz %u\n", dirent.name, dirent.size);
+		printk("\tfn '%s' size %lu\n", dirent.name, (size_t)dirent.size);
```
Comment (stephanosio) [inline samples/subsys/fs/littlefs/src/main.c:103]:
```suggestion
		printk("\tfn '%s' size %zu\n", dirent.name, dirent.size);
```

File: samples/subsys/fs/littlefs/src/main.c [hunk 4]
```diff
@@ -149,9 +149,9 @@ void main(void)
 			printk("End of files\n");
 			break;
 		}
-		printk("  %c %u %s\n",
+		printk("  %c %lu %s\n",
```
Comment (stephanosio) [inline samples/subsys/fs/littlefs/src/main.c:152]:
```suggestion
		printk("  %c %zu %s\n",
```

File: samples/subsys/fs/littlefs/src/main.c [hunk 5]
```diff
@@ -149,9 +149,9 @@ void main(void)
...
 		       (ent.type == FS_DIR_ENTRY_FILE) ? 'F' : 'D',
-		       ent.size,
+		       (size_t)ent.size,
```
Comment (stephanosio) [inline samples/subsys/fs/littlefs/src/main.c:154]:
```suggestion
		       ent.size,
```

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: samples/subsys/fs/littlefs/src/main.c (hunk 2 above)
Comment (de-nordic) [reply samples/subsys/fs/littlefs/src/main.c:77]: Everywhere else yes, but here no: the storage_dev for LittleFS, unfortunately, carries identifier of storage partition which is actually unsigned int: https://github.com/zephyrproject-rtos/zephyr/blob/30b4c29aaf5df9ed7d236394218ecd9b2f96ccef/subsys/fs/littlefs_fs.c#L566, https://github.com/zephyrproject-rtos/zephyr/blob/30b4c29aaf5df9ed7d236394218ecd9b2f96ccef/subsys/fs/littlefs_fs.c#L804)

File: samples/subsys/fs/littlefs/src/main.c (hunk 4 above)
Comment (de-nordic) [reply samples/subsys/fs/littlefs/src/main.c]: Done

File: samples/subsys/fs/littlefs/src/main.c (hunk 5 above)
Comment (de-nordic) [reply samples/subsys/fs/littlefs/src/main.c]: Done

File: samples/subsys/fs/littlefs/src/main.c (hunk 3 above)
Comment (de-nordic) [reply samples/subsys/fs/littlefs/src/main.c]: Done

File: samples/subsys/fs/littlefs/src/main.c (hunk 2 above)
Comment (stephanosio) [reply samples/subsys/fs/littlefs/src/main.c:77]: In that case:

```suggestion
		printk("FAIL: mount id %" PRIuPTR " at %s: %d\n",
		       (uintptr_t)mp->storage_dev, mp->mnt_point,
```

Comment (de-nordic) [reply samples/subsys/fs/littlefs/src/main.c]: Done

Comment (de-nordic) [reply samples/subsys/fs/littlefs/src/main.c]: @stephanosio Argh. The clang does not want to play nice with this. Will go back to (unsigned long long) as safe option and will mark this as TODO to fix when the clang gets older.

Comment (stephanosio) [reply samples/subsys/fs/littlefs/src/main.c]:
```
error: format specifies type 'unsigned int' but the argument has type 'uintptr_t' (aka 'unsigned long') [-Werror,-Wformat]
```

Hold on, something is not right there ... we should not be seeing that warning.

Comment (stephanosio) [reply samples/subsys/fs/littlefs/src/main.c]: This goes back to #16645, where we defined `__INTPTR_TYPE__` and `__UINTPTR_TYPE__` as `long int` and `long unsigned int`, respectively. These are used by the GCC and Clang headers to typedef `intptr_t` and `uintptr_t`.

https://github.com/zephyrproject-rtos/zephyr/blob/feb0e9f766b1328b45666e5e71134c34ae17b742/include/toolchain/zephyr_stdint.h#L47-L50

By default, x86-32 Clang defines `__INTPTR_TYPE__` as `int`, but we override that to `long int` in `zephyr_stdint.h`, so our `intptr_t` becomes `long int` (aka. `long`), and the Clang `inttypes.h` for x86-32 still defines `PRIuPTR` as `%u` and not `%lu`, so we get this warning/error.

This is yet another type system-related bug and needs to be fixed. I will create an issue for this.

Meanwhile, please feel free to go with your previous implementation for now (would be nice to add a `FIXME` comment though).

**Review by stephanosio — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**